Backend/UnitTesting/SessionServiceUT.cs

```diff
@@ -9,37 +10,248 @@ namespace UnitTesting
+
+            // ACT
+            using (_db = tu.CreateDataBaseContext())
+            { 
+                newSession = ss.CreateSession(_db, newSession);
+                _db.SaveChanges();
+                newSession.CreatedAt = newSession.CreatedAt.AddYears(60);
+                var response = ss.UpdateSession(_db, newSession);
+                _db.SaveChanges();
+                var result = _db.Sessions.Find(expectedResult.Id);
+
+                // Assert
+                Assert.IsNotNull(response);
+                Assert.IsNotNull(result);
+                Assert.AreEqual(result.Id, expectedResult.Id);
+                Assert.AreEqual(result.UpdatedAt, expectedResult.UpdatedAt);
+                Assert.AreEqual(result.ExpiresAt, expectedResult.ExpiresAt);
+                Assert.AreNotEqual(result.CreatedAt, expectedResultTime);
+            }
+
+        }
+
+        [TestMethod]
+        public void Update_Session_NonExisting()
+        {
+            // Arrange
+            newUser = tu.CreateUserObject();
+            newSession = tu.CreateSessionObject(newUser);
+            Session expectedResult = newSession;
+
+            // ACT
+            using (_db = tu.CreateDataBaseContext())
+            {
+                var response = ss.UpdateSession(_db, newSession);
+                try
+                {
+                    _db.SaveChanges();
+                }
+                catch (Exception)
+                {
+                    // catch error
+                    // rollback changes
+                    _db.Entry(newSession).State = System.Data.Entity.EntityState.Detached;
+                }
+                var result = _db.Sessions.Find(expectedResult.Id);
+
+                // Assert
+                Assert.IsNotNull(response);
+                Assert.IsNull(result);
+            }
+        }
+
+        [TestMethod]
+        public void Get_Session_Success()
+        {
+            // Arrange 
+            newUser = tu.CreateUserObject();
+            newSession = tu.CreateSessionObject(newUser);
+            var expectedResult = newSession;
+
+            // ACT
+            using (_db = tu.CreateDataBaseContext())
+            {
+                newSession = ss.CreateSession(_db, newSession);
+                _db.SaveChanges();
+                var result = ss.GetSession(_db, newSession.Token, newUser.Id);
+
+                // Assert
+                Assert.IsNotNull(result);
+                Assert.AreEqual(expectedResult.Id, result.Id);
+            }
+        }
+
+        [TestMethod]
+        public void Get_Session_NonExisting()
+        {
+            // Arrange
+            Guid nonExistingSession = Guid.NewGuid();
+            string nonExistingToken = Guid.NewGuid().ToString();
+            Session expectedResult = null;
+
+            // Act
+            using (_db = tu.CreateDataBaseContext())
+            {
+                var result = ss.GetSession(_db, nonExistingToken, nonExistingSession);

```

Comment:
Here as well, nonExistingSession is actually a userID param, needs to be removed